sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/OperationPoliciesTest.java

```diff
@@ -0,0 +1,772 @@
+            .addOperationPolicy(policy);
+        clientBuilders[2] = new CosmosClientBuilder().endpoint(TestConfigurations.HOST)
+            .credential(credential)
+            .addOperationPolicy((cosmosOperationDetails) -> {
+                CosmosDiagnosticsContext cosmosDiagnosticsContext = cosmosOperationDetails.getDiagnosticsContext();
+                String operationType = cosmosDiagnosticsContext.getOperationType();
+                String spanName = cosmosDiagnosticsContext.getSpanName();
+                CosmosRequestOptions cosmosRequestOptions = new CosmosRequestOptions();
+                createReadDeleteBatchEtcOptions(operationType, spanName, cosmosRequestOptions);
+                createQueryReadAllItemsOptions(operationType, spanName, cosmosRequestOptions);
+                createReadManyOptions(spanName, cosmosRequestOptions);
+                cosmosOperationDetails.setCommonOptions(cosmosRequestOptions);
+            }).addOperationPolicy((cosmosOperationDetails) -> {
+                CosmosDiagnosticsContext cosmosDiagnosticsContext = cosmosOperationDetails.getDiagnosticsContext();
+                String operationType = cosmosDiagnosticsContext.getOperationType();
+                String spanName = cosmosDiagnosticsContext.getSpanName();
+                CosmosRequestOptions cosmosRequestOptions = new CosmosRequestOptions();
+                createBulkOptions(operationType, spanName, cosmosRequestOptions);
+                createChangeFeedOptions(spanName, cosmosRequestOptions);
+                cosmosOperationDetails.setCommonOptions(cosmosRequestOptions);
+            });
+        return clientBuilders;
+    }
+
+    @BeforeClass(groups = {"fast"}, timeOut = SETUP_TIMEOUT)
+    public void before_OperationPoliciesTest() {
+        assertThat(this.client).isNull();
+        this.client = getClientBuilder().buildAsyncClient();
+        CosmosAsyncContainer asyncContainer = getSharedMultiPartitionCosmosContainer(this.client);
+        container = client.getDatabase(asyncContainer.getDatabase().getId()).getContainer(asyncContainer.getId());
+    }
+
+    @AfterClass(groups = {"fast"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        assertThat(this.client).isNotNull();
+        this.client.close();
+    }
+
+    @AfterMethod()
+    public void afterMethod() {
+        changeProperties(initialOptions);
+    }
+
+    @DataProvider(name = "changedOptions")
+    private String[][] createChangedOptions() {
+        return new String[][] {
+            { "8", ConsistencyLevel.SESSION.toString().toUpperCase(), "true", "false", "true", "defaultChanged", "1000", "true", "West US 2", "4", "200", "400", "100", "false", "false", "20", "QueryNameChanged" },
+            { "4", ConsistencyLevel.EVENTUAL.toString().toUpperCase(), "false", "true", "true", "defaultChanged", "1000", "true", "West US 2", "4", "200", "400", "100", "false", "false", "20", "QueryNameChanged" },
+            initialOptions
```

Comment:
`initialOptions` is always applied once + mutation for every test. Is it needed in `DataProvider`?

Reply:
discussed offline.